File: src/main/java/no/nav/bidrag/beregn/barnebidrag/beregning/BarnebidragBeregning.java
package no.nav.bidrag.beregn.barnebidrag.beregning;

import java.util.List;
import no.nav.bidrag.beregn.barnebidrag.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.barnebidrag.bo.ResultatBeregning;

public interface BarnebidragBeregning {

  List<ResultatBeregning> beregn(GrunnlagBeregning grunnlagBeregning);

  static BarnebidragBeregning getInstance(){
    return new BarnebidragBeregningImpl();
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/barnebidrag/beregning/BarnebidragBeregning.kt
package no.nav.bidrag.beregn.barnebidrag.beregning

import no.nav.bidrag.beregn.barnebidrag.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.barnebidrag.bo.ResultatBeregning


fun interface BarnebidragBeregning {
    fun beregn(grunnlag: GrunnlagBeregning): List<ResultatBeregning>

    companion object {
        fun getInstance(): BarnebidragBeregning {
            return BarnebidragBeregningImpl()
        }
    }
}
